types/enzyme/enzyme-tests.tsx

```diff
@@ -640,6 +663,21 @@ function ReactWrapperTest() {
         boolVal = reactWrapper.find('.my-button').hasClass('disabled');
     }
 
+    function test_invoke() {
+        let is_bool: boolean;
+        is_bool = reactWrapper.invoke('functionProp')(); // no args
```

Comment:
Assignability can be a weak type-level assertion. Invoke could start returning any, for example, and all these tests would pass. Can you try `// $ExpectType number` instead for these? I'm not sure where the magic comment can be placed (above, same line, etc)

Reply:
After working with the aboves changes, I've started pulling out the "invoked" assignment tests. After thinking about it, these tests just prove that `functionProp` works as expect, not `invoke`.   I wrote these originally after looking at the enzyme docs for `invoke`. Being inspired by the `(...args)` in the docs, it felt necessary, but I'm feeling less so now.

Reply:
is "$ExpectType" a TS thing, or a DT tests thing?

Reply:
Seems to be part of [dtslint](https://github.com/microsoft/dtslint).